plugins/autowarmth.koplugin/main.lua

```diff
@@ -0,0 +1,780 @@
+--[[--
+@module koplugin.autowarmth
+
+Plugin for setting screen warmth based on the sun position and/or a time schedule
+]]
+
+local Device = require("device")
+
+local ConfirmBox = require("ui/widget/confirmbox")
+local DoubleSpinWidget = require("/ui/widget/doublespinwidget")
+local DeviceListener = require("device/devicelistener")
+local Dispatcher = require("dispatcher")
+local FFIUtil = require("ffi/util")
+local InfoMessage = require("ui/widget/infomessage")
+local InputDialog = require("ui/widget/inputdialog")
+local Font = require("ui/font")
+local SpinWidget = require("ui/widget/spinwidget")
+local SunTime = require("suntime")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local _ = require("gettext")
+local T = FFIUtil.template
+local Screen = require("device").screen
+local util = require("util")
+
+local activate_sun = 1
+local activate_schedule = 2
+local activate_closer_noon = 3
+local activate_closer_midnight =4
+
+local midnight_index = 11
+
+local device_max_warmth = Device:hasNaturalLight() and Device.powerd.fl_warmth_max or 100
+local device_warmth_fit_scale = device_max_warmth / 100
+
+local function frac(x)
+    return x - math.floor(x)
+end
+
+local AutoWarmth = WidgetContainer:new{
+    name = "autowarmth",
+    easy_mode = G_reader_settings:isTrue("autowarmth_easy_mode") or false,
```

Comment:
Generally an easy mode is the default with the assumption that those looking for more will find advanced mode quickly enough. Plus I suspect people may not like toggling "easy mode". But I'm not sure if there's any research on it.
```suggestion
    easy_mode = G_reader_settings:nilOrTrue("autowarmth_easy_mode"),
```

---

@NiLuJe Any comments wrt <https://github.com/koreader/koreader/pull/7371>?

Reply:
Agree on making "simple" the default.

I'll leave the LuaSettings decision to @zwim, but I am indeed not a fan of the magic `nil` defaults ;).